src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -93,22 +96,23 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):#pylint:
 register_cli_argument('vm container create', 'agent_vm_size', completer=get_vm_size_completion_list)
 
 register_cli_argument('vm capture', 'overwrite', action='store_true')
+register_cli_argument('vm nic', 'vm_name', existing_vm_name, id_part=None)
 register_cli_argument('vm nic', 'nic_ids', multi_ids_type)
 register_cli_argument('vm nic', 'nic_names', multi_ids_type)
 register_cli_argument('vm diagnostics', 'vm_name', arg_type=existing_vm_name, options_list=('--vm-name',))
 register_cli_argument('vm diagnostics set', 'storage_account', completer=get_resource_name_completion_list('Microsoft.Storage/storageAccounts'))
 
-register_cli_argument('vm extension', 'vm_extension_name', name_arg_type, completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachines/extensions'))
-register_cli_argument('vm extension', 'vm_name', arg_type=existing_vm_name, options_list=('--vm-name',))
+register_cli_argument('vm extension', 'vm_extension_name', name_arg_type, completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachines/extensions'), id_part='child_name')
+register_cli_argument('vm extension', 'vm_name', arg_type=existing_vm_name, options_list=('--vm-name',), id_part='name')
 register_cli_argument('vm extension', 'no_auto_upgrade', action='store_true')
 
 register_cli_argument('vm extension image', 'image_location', options_list=('--location', '-l'))
 register_cli_argument('vm extension image', 'publisher_name', options_list=('--publisher',))
 register_cli_argument('vm extension image', 'type', options_list=('--name', '-n'))
 register_cli_argument('vm extension image', 'latest', action='store_true')
 
-register_cli_argument('vmss extension', 'extension_name', name_arg_type, id_part='name', help='Name of extension')
-register_cli_argument('vmss extension', 'vmss_name', help='Scale set name')
+register_cli_argument('vmss extension', 'extension_name', name_arg_type, help='Name of the extension.')
+register_cli_argument('vmss extension', 'vmss_name', help='Scale set name', id_part=None)
```

Comment:
Yes because "name_arg_type" has "id_part='name'" by default.

---
In reply to: [77102525](https://github.com/Azure/azure-cli/pull/803#discussion_r77102525) [](ancestors = 77102525)